app/models/user.rb

```diff
@@ -0,0 +1,10 @@
+class User < ApplicationRecord
+  before_save { self.email = email.downcase }
+  validates :name, presence: true, length: { maximum: 50 }
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  validates :email, presence: true, length: { maximum: 255 },
+                    format: { with: VALID_EMAIL_REGEX },
+                    uniqueness: true
+  has_secure_password
+  validates :password, presence: true, length: { minimum: 6 }
```

Comment:
` presence: true` có tác dụng gi em

Reply:
argument presence: true có tác dụng khi em nhập vào một chuỗi rỗng nó sẽ không chấp nhận
VD:
user = User.new(name: "", email: "test@example.com")
dùng hàm valid? để check object có hợp lệ không thì hàm sẽ return false: user.valid? -> false
cụ thể là user.errors.full_messages -> Name can't be blank

Reply:
À ở đây hàm validates nhận argument là hash nhưng không cần {} vì hash mà là argument cuối truyền vào hàm thì không cần {}